src/cached-fetcher/index.ts

```diff
@@ -14,6 +14,12 @@ const setCachedPromise = <Result>(
 ): void => {
   const cacheKey = key;
 
+  // Ensure any existing timeout is cleared before setting a new one
+  const existingEntry = cache.get(cacheKey);
+  if (existingEntry?.timeoutId !== undefined) {
```

Comment:
maybe just 
```suggestion
  if (!isEmpty(existingEntry?.timeoutId)) {
```

import {isEmpty} from the other utils. This ensures null, undefined scenarios